internal/controller/drclusterconfig_controller.go

```diff
@@ -272,12 +288,71 @@ func (r *DRClusterConfigReconciler) processCreateOrUpdate(
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	if err := r.validateS3Profiles(ctx, log, drCConfig); err != nil {
+		log.Info("Reconcile error", "error", err)
+		setDRClusterConfigConfigurationProcessedCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
+			err.Error(), metav1.ConditionFalse, DRClusterConfigS3Unreachable)
+
+		return ctrl.Result{Requeue: true}, err
+	}
+
 	setDRClusterConfigConfigurationProcessedCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
```

Comment:
Move this above checking the s3 rechability, such that processed condition is set to the right value independent of s3 validation.